### DISCUSSION TITLE
Walkthroughs with purchase links?

### DISCUSSION THREAD

**Discussion by benallfree** [theyosh/TerrariumPI#824] [Q&A] [answered]
I'm looking for a TerrariumPi walkthrough that has some amazon purchase links. I've done minimal Pi engineering but am a very experienced programmer. The hardware stuff is new to me.

We are trying to build an enclosure that adjusts day/night heat lamps, maintains humidity and temp, and cycles in new air via an intake and exhaust fan. I'm not sure how to make all this work together, so I was hoping there might be a walkthrough somewhere.

**Comment by theyosh** [comment, marked as answer]
Hi,

now, I am afraid there is no walk through. As I support 30 relays https://theyosh.github.io/TerrariumPI/hardware/ and even more sensors. So there are to many options to make a walk through.

And to make it harder, some hardware will exclude other hardware due to the way it is connected to the PI, or used connections.

And to make it even more harder, not all hardware is available in all countries.

But it looks like you need a lot or relays. So looking for devices like https://theyosh.github.io/TerrariumPI/hardware/relay/energenie-usb/ can work. 4 relays in 1 device.

You need some sensors. I would go for 1wire setup, as you can plug a lot of sensors to the Pi. But this requires wiring your TerrariumPI. Alternative, there are some bluetooth sensors, but they are battery operated, and needs to replaced once a time.

So do you have a terrarium at the moment? And it you want to build a complete system, always order extra parts. After a few years, some sensors / relays or other hardware will not be available anymore. So keeping spares is smart thing to do,

That is what I can say now. There is unfortunately not a simple walk through.
